site/src/scaffold/scripts/popover.ts

```diff
@@ -0,0 +1,72 @@
+export function initPopovers(container: HTMLElement) {
+	container.addEventListener('click', (event: Event) => {
+		let targetPopover: HTMLDetailsElement | null;
+		targetPopover =
+			event.target instanceof Element &&
+			(event.target.closest('details.popover') as HTMLDetailsElement)
+				? (event.target.closest('details.popover') as HTMLDetailsElement)
+				: null;
+
+		if (!targetPopover) {
+			targetPopover =
+				event.target instanceof Element &&
+				event.target.shadowRoot &&
+				event.target.shadowRoot.activeElement &&
+				(event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					? (event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					: null;
+		}
+
+		if (!targetPopover) {
+			return;
+		}
+
+		const allPopovers: HTMLDetailsElement[] = Array.from(
+			container.querySelectorAll('details.popover')
+		);
+
+		if (!targetPopover.open) {
```

Comment:
question: can this be placed above the querySelection of `allPopovers`? so that `if (!targetPopover.open)` can be removed. Just reduces the amount of nesting.

```ts
if (targetPopover.open) {
  return;
}

Reply:
Sure 👍🏻

Reply:
Updated